great_expectations/execution_engine/execution_engine.py

```diff
@@ -184,11 +190,12 @@ def __init__(
             if key in self.recognized_batch_spec_defaults
         }
 
-        self._batch_data_dict = {}
+        self._batch_manager = BatchManager(execution_engine=self)
```

Comment:
A naive question. Thoughts on pulling the `BatchManager` totally out of `ExecutionEngine`? I know a goal of this effort is to "unbundle" things so I'm wondering if letting a caller manage when it needs a batch manager is helpful/realistic.

Reply:
I think this could be a good idea, especially since so many of the other properties reference attributes of the BatchManager via the ExecutionEngine

Reply:
@billdirks I am not sure how that would work -- `ExecutionEngine` needs services from `BatchManager` (`ExecutionEngine` used to have all of the relevant logic inside itself, but now it is encapsulated in `BatchManager`).

Reply:
@billdirks I also do not see how we would obtain a handle to `BatchManager` without instantiating it in the `ExecutionEngine` parent class, other than creating a singleton; but we already have `DataContext` as the singleton, and we do not want to create another one right now (moreover, `ExecutionEngine` does not use `DataContext`).  So would prefer to leave it like it is for now (it is congruent with the existing code).  Thanks.